File: examples/DesignPatterns/FactoryMethod/src/main/java/dispatchers/impl/StringDispatcher.java
package dispatchers.impl;

import dispatchers.Dispatcher;
import models.Action;
import models.ActionResult;

public class StringDispatcher implements Dispatcher<String> {
    private int dispatcherID;

    public StringDispatcher(int dispatcherID) {
        this.dispatcherID = dispatcherID;
    }

    @Override
    public ActionResult<String> dispatch(Action<String> action) {
        return new ActionResult<>(action.getType(), action.getPayload(), this.dispatcherID);
    }

    public int getDispatcherID() {
        return dispatcherID;
    }

    public void setDispatcherID(int dispatcherID) {
        this.dispatcherID = dispatcherID;
    }
}

File: examples/DesignPatterns/FactoryMethod/typescript/src/dispatchers/impl/StringDispatcher.ts
import { Action, ActionResult } from '../../models';
import { Dispatcher } from '../Dispatcher';

export class StringDispatcher implements Dispatcher<string> {
  private _dispatcherID: number;

  constructor(dispatcherID: number) {
    this._dispatcherID = dispatcherID;
  }

  public dispatch(action: Action<string>): ActionResult<string> {
    return new ActionResult(action.type, action.payload, this._dispatcherID);
  }

  public get dispatcherID(): number {
    return this._dispatcherID;
  }
  public set dispatcherID(value: number) {
    this._dispatcherID = value;
  }
}
